src/Microsoft.AspNet.Authentication.Facebook/FacebookAuthenticationHandler.cs

```diff
@@ -51,7 +51,12 @@ public FacebookAuthenticationHandler(HttpClient httpClient)
 
         protected override async Task<AuthenticationTicket> CreateTicketAsync(ClaimsIdentity identity, AuthenticationProperties properties, OAuthTokenResponse tokens)
         {
-            var endpoint = Options.UserInformationEndpoint + "?access_token=" + UrlEncoder.UrlEncode(tokens.AccessToken);
+            var endpoint = Options.UserInformationEndpoint;
+            var accessToken = "access_token=" + UrlEncoder.UrlEncode(tokens.AccessToken);            
+                        
+            endpoint += endpoint.Contains("?") ? "&" : "?";
```

Comment:
@PinpointTownes Yeah, I was thinking about using a helper to build the query string, but wasn't sure about @HaoK 's thoughts on the subject since the other Auth providers don't do anything like that. I was exercising some caution not to move so much code. @HaoK , would you prefer that I use `QueryHelpers`?